samcli/cli/types.py

```diff
@@ -474,3 +475,61 @@ def convert(self, value, param, ctx):
         LOG.debug("Converting provided %s option value to Enum", param.opts[0])
         value = super().convert(value, param, ctx)
         return self.enum(value)
+
+
+class SyncWatchExcludeType(click.ParamType):
```

Comment:
Apologies that I missed in the original review but what happens if user provides multiple exclude folders for same function? Is it going to work or one will replace the other?

Reply:
If this is provided multiple times, they'll all be excluded. If a user passes in:
- `--watch-exclude Function=file1`
- `--watch-exclude Function=file2`
- `--watch-exclude Function=file3`
all three files will be excluded from being watched.

Same thing with folders, in the case that the user passes in
- `--watch-exclude Function=folder/child1`
- `--watch-exclude Function=folder`
they both will still be passed in, however this is just the same as passing in `--watch-exclude Function=folder`